ui/hooks/useTransactionDisplayData.js

```diff
@@ -121,18 +128,33 @@ export function useTransactionDisplayData(transactionGroup) {
   // This value is used to determine whether we should look inside txParams.data
   // to pull out and render token related information
   const isTokenCategory = TOKEN_CATEGORY_HASH[type];
+  const { tokenSymbol, decimals, toAddress } = useAssetDetails(
+    recipientAddress,
+    senderAddress,
+    initialTransaction?.txParams?.data,
+    isTokenCategory,
+  );
+  const customToken = { address: toAddress, symbol: tokenSymbol, decimals };
 
   // these values are always instantiated because they are either
   // used by or returned from hooks. Hooks must be called at the top level,
   // so as an additional safeguard against inappropriately associating token
   // transfers, we pass an additional argument to these hooks that will be
   // false for non-token transactions. This additional argument forces the
   // hook to return null
-  const token =
-    isTokenCategory &&
-    knownTokens.find(({ address }) =>
-      isEqualCaseInsensitive(address, recipientAddress),
-    );
+
+  let token = null;
+  if (isTokenCategory) {
+    token =
+      knownTokens.find(({ address }) =>
+        isEqualCaseInsensitive(address, recipientAddress),
+      ) ||
+      detectedTokens.find(({ address }) =>
+        isEqualCaseInsensitive(address, recipientAddress),
+      ) ||
+      tokenList[recipientAddress.toLowerCase()] ||
+      customToken;
+  }
 
```

Comment:
Will be nice to have test coverage for above hook.

Reply:
Test is added by adding a new set of data to transaction-data.json